Core/Registry/RegistryManager.cs

```diff
@@ -449,8 +449,20 @@ public CkanModule GenerateModpack(bool recommends = false, bool with_versions =
                 download_content_type = "application/zip",
             };
 
-            List<RelationshipDescriptor> mods = registry.Installed()
-                .Where(mod => !(mod.Value is ProvidesModuleVersion || mod.Value is UnmanagedModuleVersion))
+            List<RelationshipDescriptor> mods = registry.Installed(false)
+                .Where(kvp => !(kvp.Value is UnmanagedModuleVersion))
+                .Where(kvp => {
+                    // Skip unavailable modules (custom .ckan files)
+                    try
+                    {
+                        var avail = registry.LatestAvailable(kvp.Key, null, null);
```

Comment:
I just found out that this is apparently not a 100% way to detect custom modules installed from a .ckan:
Before installing, the module is added to the `available_modules`:
https://github.com/KSP-CKAN/CKAN/blob/a1b6d6a22791d51d41103e92cb76a30f76ee3da5/GUI/Main/Main.cs#L527-L531

And it is removed the next time the registry is updated from the repository. Until then the custom module will be in the dependency group by default.

We should check whether it is actually needed for the installation flow to add the local module to `available_modules` in `installFromckanToolStripMenuItem_Click`.

Reply:
Yeah, I've been suspicious of that aspect of GUI for a long time. Notably, `ckan install -c whatever.ckan` does **not** add the module to `available_modules`.

I just tested with it removed and didn't notice any problems after installing a custom module. Looks OK to change.

Reply:
Yep same, and also took a short dive into the installation/relationship resolving code, and couldn't find a place where it would expect the new modules to be registered in the available modules.